test/psim/core/configuration_test.cpp

```diff
@@ -34,7 +34,7 @@ TEST(Configuration, TestGet) {
   auto const config = psim::Configuration::make(file);
 
   // Test a proper access
-  ASSERT_EQ(config.get("test.integer")->get<psim::Integer>(), 1);
+  ASSERT_EQ(config.get("test.integer")->template get<psim::Integer>(), 1);
```

Comment:
Why is this valid C++? I'm confused by the syntax (why is the space between template and get allowed?)

Reply:
That's how the `template` and `typename` keywords are actually intended to work. Here's a cool explanation about why the parser requires them: https://stackoverflow.com/questions/610245/where-and-why-do-i-have-to-put-the-template-and-typename-keywords

Essentially though, they're required sometimes to fix ambiguity. Although in this case it seems only apple clang was complaining on not GCC.